src/runtime_src/core/common/sensor.cpp

```diff
@@ -250,22 +249,31 @@ read_data_driven_thermals(const std::vector<xq::sdm_sensor_info::data_type>& out
 }
 
 static ptree_type
-read_data_driven_mechanical(std::vector<xq::sdm_sensor_info::data_type> output)
+read_data_driven_mechanical(std::vector<xq::sdm_sensor_info::data_type>& output,
+                            std::vector<xq::sdm_sensor_info::data_type>& temp)
 {
   ptree_type root;
   ptree_type pt;
   ptree_type fan_array;
 
   // iterate over output data, store it into property_tree
-  for(const auto& tmp : output)
+  for (const auto& tmp : output)
   {
     pt.put("location_id", tmp.label);
     pt.put("description", tmp.label);
     pt.put("speed_rpm", tmp.input);
-    pt.put("critical_trigger_temp_C", "N/A");
     pt.put("is_present", "true");
-    fan_array.push_back({"", pt});
   }
+  // iterate over output data, store it into property_tree
+  for (const auto& tmp : temp)
+  {
+    if (boost::iequals(tmp.label, "Vccint Temp"))
```

Comment:
This logic is confusing, what relationship does "Vccint Temp" have with "critical_trigger_temp_C".

Reply:
We discussed with firmware team yesterday, and they suggested not to use Vccint Temp for fan's temperature. And also confirmed this data is not available for versal.